alephium/lib/deployment.ts

```diff
@@ -0,0 +1,389 @@
+import {
+  BuildDeployContractTx,
+  BuildExecuteScriptTx,
+  NodeProvider,
+  Contract,
+  Script,
+  SignerWithNodeProvider,
+  groupOfAddress,
+  Account,
+  publicKeyFromPrivateKey,
+  addressFromPublicKey,
+  signatureEncode,
+  Project,
+  setCurrentNodeProvider,
+  Token
+} from "@alephium/web3"
+import { getWalletFromMnemonic } from "@alephium/sdk"
+import path from "path"
+import { ec as EC } from 'elliptic'
+import { waitTxConfirmed } from "./utils"
+import fs, { promises as fsPromises } from "fs"
+import * as cryptojs from 'crypto-js'
+
+export interface Network {
+  nodeUrl: string
+  mnemonic: string
+  scripts: string[] // script file path, execute by order
+  deploymentFile: string
+}
+
+export type NetworkType = "mainnet" | "testnet" | "devnet"
+
+export interface Configuration {
+  sourcePath?: string
+  artifactPath?: string
+
+  networks: Record<NetworkType, Network>
+}
+
+type DeployContractParams = Omit<BuildDeployContractTx, 'signerAddress'>
+
+export interface ExecutionResult {
+  fromGroup: number
+  toGroup: number
+  txId: string
+  blockHash: string
+  codeHash: string
+  attoAlphAmount?: string
+  tokens?: Record<string, string>
+}
+
+export interface DeployContractResult extends ExecutionResult {
```

Comment:
Remark: we will check what fields are missing compared to Hardhat's DeployContractResult. At a first glance, Hardhat has gas information.

Could create Github issues for this.